### DISCUSSION TITLE
Include languages for private organization repos?

### DISCUSSION THREAD

**Discussion by thorpelawrence** [lowlighter/metrics#190] [Q&A] [answered]
Hi there!

Just wondering, is it possible to get language metrics for all repos you have access to, that aren't owned directly by you - for example, if I 
contribute to some private repos in an organization I own but I want these languages to show on my profile?

My token has scopes: `repo`, `org:read`

For example

```graphql
organizations(first: 10) {
      nodes {
        repositories(first: 10) {
          nodes {
            languages(first: 10) {
              edges {
                size
                node {
                  color
                  name
                }
              }
            }
          }
        }
      }
    }
```

**Comment by lowlighter** [comment, marked as answer]
Hello!

Yes, you can set the `repositories_affiliations` option:
https://github.com/lowlighter/metrics/blob/b17eba9675cf992af68821d2723c8ed7d032b6c0/source/plugins/base/metadata.yml#L37-L47

Using `""` on it will remove entirely the `affiliations` constraint from the query. 

I don't have that much experience with GitHub organizations so I'm not entirely sure, but it may also include repositories where you're a  member of said organization but have not actually contributed too. So you could also try to set it to `collaborator` level.

*(note that some plugins re-use same the data extracted by base plugin, which is the case for the languages plugin, so that's why even though it's a base plugin option, it'll affect the languages stats)*

Hope this helps 🙂

**Reply by thorpelawrence** [reply]
I will have a look, thanks for your help! 🙂

**Reply by Napolitain** [reply]
@thorpelawrence did you try? I haven't, but I think normally "owner" would work for the repositories you own in a organization no?

**Reply by thorpelawrence** [reply]
@Napolitain I was able to get it to work, but it was either collecting languages for every repo, in all orgs I belong to (which made my results not very relevant to me, since they were languages used in the org that I don't use much), or they didn't make much difference (I'm guessing since there's only a few I actually 'own' in orgs)
